components/org.wso2.carbon.identity.scim2.common/src/main/java/org/wso2/carbon/identity/scim2/common/impl/SCIMRoleManager.java

```diff
@@ -482,6 +486,23 @@ private void doUpdatePermissions(Role oldRole, Role newRole) throws BadRequestEx
         return removedIDs;
     }
 
+    private boolean hasPermissionsChanged(List<String> oldRolePermissions, List<String> newRolePermissions) {
+
+        if (newRolePermissions == null) {
```

Comment:
what happens if the new role doesn't have any permission?

Reply:
It comes as an empty array.